### PR TITLE
Module 1

### PR DESCRIPTION
**Pull request by alanjknight**
Please make sure you are raising this PR against your own repository and not the original. If it says https://github.com/CorndelWithSoftwire in the search bar right now, you are in the wrong place!

Please also don't forget to submit the PR's url to Aptem afterwards (by clicking on the corresponding exercise component, pasting the url in the textbox that appears and then pressing the finish button).

### CODE REVIEW

File: todo_app/app.py
Comment (JackMeadDev) [inline todo_app/app.py]: Worth noting that we can simplify this quite a lot by taking advantage of the default argument in `request.args.get`, e.g.
we can replace this combo of lines:
```python
    sort_col="id"
    if request.args.get("sort_col"):
        sort_col=request.args.get("sort_col")

```
with:
```python
sort_col = request.args.get("sort_col", "id")
```

File: todo_app/app.py
Comment (JackMeadDev) [inline todo_app/app.py]: Similarly, we can condense this a little:
```python
sort_reverse = (sort_reverse == "True")
```

File: todo_app/templates/index.html
Comment (JackMeadDev) [inline todo_app/templates/index.html:24]: If we change the `type` here to "date" then most browsers will provide some default assistance, e.g. letting the user select the date with a calendar. The default date format is "dd/mm/yyyy" though so we'd need to slightly adjust our server to expect that instead

File: todo_app/app.py
Comment (JackMeadDev) [inline todo_app/app.py]: It's good practice to get into the habit of clearing out these print statements that are useful for development debugging before committing the code

File: todo_app/templates/index.html
Comment (JackMeadDev) [inline todo_app/templates/index.html:11]: Is there a reason you've chosen to switch off validation for this form? This will prevent us taking advantage of built in browser support, like having the browser clearly highlight required fields to save the user even having to hit our server at all before we verify their inputs

**Review by JackMeadDev — APPROVED** [review]
This looks good, I was able to easily create/view/adjust my To-dos on your app following the docs, and nice work on the stretch goals! In future try to hold off merging until we've reviewed the code, as it's easier to review if all of the relevant parts of the module are in a single PR, but not a problem here.

I've left some thoughts below, but I'm happy for you to merge this when you're ready

File: todo_app/app.py
Comment (alanjknight) [reply todo_app/app.py]: Thanks, Jack - updated as shown.

File: todo_app/app.py
Comment (alanjknight) [reply todo_app/app.py]: Thanks, Jack - updated.

File: todo_app/app.py
Comment (alanjknight) [reply todo_app/app.py]: Damn print statements! - Removed!

File: todo_app/templates/index.html
Comment (alanjknight) [reply todo_app/templates/index.html:11]: On the final point about switching off the validation on the form, I did this as I wanted to do all the validation at the back end (Just to ensure I did more Python, more than anything else!).  Also wasn't sure that all browsers would behave in the same way.  Get your point though - would be more efficient to have the front end do some validation.

**Pull request merged**
